presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -628,7 +628,7 @@ private static DynamicFiltersResult createDynamicFilters(
         {
             Map<String, VariableReferenceExpression> dynamicFilters = ImmutableMap.of();
             List<RowExpression> predicates = ImmutableList.of();
-            if (node.getType() == INNER) {
+            if (node.getType() == INNER || node.getType() == RIGHT) {
```

Comment:
Why not left join? Are you assuming right side is going to be the build?

Reply:
yes, dynamic filtering applies to broadcast join, and the right side is the build.
since dynamic filtering is generating predicates and propagate to the probe side, right join is fine, but left join might not be correct.

Reply:
Looks like you are assuming that RIGHT join will never be rewritten to LEFT due to some other optimizations? I'm just wondering if this is useful. Please add a real usecase and also more comprehensive testing (including hive end to end) and also a session param to control this feature (with defualt off) if we are convinced this is a good thing to do.

Reply:
I do not think RIGHT will be rewritten to LEFT when it comes here, since:
1. dynamic filtering only applies to broadcast join, we already have session properties to turn it on/off
2. the code here is a branch assuming dynamic filtering is on
3. cost based optimizer is not used
SystemSessionProperties.isEnableDynamicFiltering is the session property to control dynamic filtering
The use case is:
`SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey`
The motivation to work on this feature it, when I was implementing comparison operator support for dynamic filtering, was trying to start from simple, so assuming left and right to be variable, and only for inner join. Actually, find support right join and expression for probe side is just a few lines changes, so submitted this PR.